### PR TITLE
[Inductor] Add register_lowering for aten.arange.start_step (#179028)

### PR DESCRIPTION
**Pull request by nandesuka**
Summary:

## Summary

`aten.arange.start_step` only existed in Inductor's decomposition table (decomposed to `prims.iota`), but was NOT registered in the `lowerings` dict. This caused `KeyError` when any code tried to access `lowerings[aten.arange.start_step]`.

This fixes 176 opinfo e2e test failures for `arange` on MTIA, where `arange_int_lowering.py` (D98064707) delegates to `lowerings[aten.arange.start_step]`.

### Root Cause

D98064707 added `arange_int_lowering.py` which registers a lowering for `torch.ops.mtia.arange_int` and delegates to `lowerings[aten.arange.start_step]`. However, upstream Inductor only registers `aten.arange` in its **decomposition** table (decomposed to `prims.iota` at `lowering.py:3453`), not in the **lowerings** dict. The unit tests masked this by injecting a mock via `patch.dict(lowerings, {aten.arange.start_step: mock_arange})`.

### Fix

Add `register_lowering(aten.arange.start_step)` that delegates to the existing `prims.iota` lowering (which generates `Pointwise.create` with `ops.index_expr`). This makes the key available in the `lowerings` dict for any downstream code.

### Changed Files

- `fbcode/caffe2/torch/_inductor/lowering.py` — add `arange_start_step` lowering after `iota`
- `xplat/caffe2/torch/_inductor/lowering.py` — mirror
- `fbcode/caffe2/test/inductor/test_torchinductor.py` — add `test_arange7` for integer arange via `aten.arange.start_step`
- `xplat/caffe2/test/inductor/test_torchinductor.py` — mirror

Test Plan:
- `test_arange7` in `test_torchinductor.py` — verifies `aten.arange.start_step` with integer dtype
- Existing `test_arange3` already calls `aten.arange.start_step` directly — passes with new lowering
- MTIA `test_arange_int_lowering` unit tests — delegation to `lowerings[aten.arange.start_step]` no longer raises KeyError
- Opinfo e2e: `arange; 1; 0; int, 2` should compile successfully with auto DS

Reviewed By: blaine-rister

Differential Revision: D98755321

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -3191,6 +3191,15 @@ def fn(x):
...
         self.common(fn, (make_arg(1, dtype=torch.int64),))
 
+    def test_arange7(self):
+        def fn(x):
+            # Test aten.arange.start_step lowering with integer dtypes
+            return x + torch.ops.aten.arange.start_step(
+                0, 10, 2, dtype=torch.int64, device=x.device
+            )
+
+        self.common(fn, (torch.zeros(5, dtype=torch.int64),), check_lowp=False)
```
Comment (blaine-rister) [inline test/inductor/test_torchinductor.py:3200]: nit: should we check that this actually generates a Triton kernel? Does `self.common` already handle this?

**Review by blaine-rister — APPROVED** [review]
LGTM, left a nit about possibly strengthening the test

File: test/inductor/test_torchinductor.py (hunk 1 above)
Comment (nandesuka) [reply test/inductor/test_torchinductor.py:3200]: yup for GPU tests it generates Triton kernels, for CPU tests it uses C++
